LLama/ChatSession.cs

```diff
@@ -494,3 +584,50 @@ in OutputTransform
         }
     }
 }
+
+/// <summary>
+/// The state of a chat session in-memory.
+/// </summary>
+public record SessionState
+{
+    /// <summary>
+    /// Saved executor state for the session in JSON format.
+    /// </summary>
+    public string ExecutorState { get; init; }
+
+    /// <summary>
+    /// Saved context state (KV cache) for the session.
+    /// </summary>
+    public State ContextState { get; init; }
+
+    /// <summary>
+    /// The input transform pipeline used in this session.
+    /// </summary>
+    public List<ITextTransform> InputTransformPipeline { get; init; } = new();
+
+    /// <summary>
+    /// The output transform used in this session.
+    /// </summary>
+    public ITextStreamTransform OutputTransform { get; init; } = new LLamaTransforms.EmptyTextOutputStreamTransform();
```

Comment:
I think we should re-consider the design here. Since the transforms are included here, it's likely telling users that the state of each transform will be well saved. However, in some cases it does not, for example if the user implement a transform with internal state, the state may be lost after re-load it from the saved state file.

Let's gather the concerns here:
1. Should the transforms be a part of the session state?
2. Is the session state expected to be able to save as file?
3. If transforms are included, how to ensure they can be correctly saved and loaded?

I'm open for these points and I'd like to provide my views here.
1. I prefer to include them but should only do so if they could be properly saved and loaded. If you finally decide to remove them, please add some warnings to let users know the potential inconsistent behavior after loading the session.
2. Yes, I think it's a common utilization of the state.
3. Maybe we could add a new interface `ISerializableTransform` with a pair of method `Serialize` and `Deserialize`, moving the responsibility of converting between (json) string and class to user. For the internally defined transforms in LLamaSharp, we could implement this interface for them.

Reply:
I generally was aiming at in-memory only, non-serializable SessionState, as there is already a way to save/load ChatSession into file.
If it's non-serializable, then just storing transform objects suffices I think.

Reply:
It's okay to keep the current design in this PR. The file saving and loading of `ChatSession` still has some problems. I'll head to it after having this PR merged.